src/libsyntax/feature_gate.rs

```diff
@@ -40,6 +40,13 @@ use symbol::{keywords, Symbol};
 use std::{env, path};
 
 macro_rules! set {
+    (const_fn) => {{
```

Comment:
It'd be nice to leave a comment here to make clear exactly why this special case occurs.